### PR TITLE
More Ghost Text Improvements

### PR DESCRIPTION
**Pull request by hediet**

### CODE REVIEW

File: src/vs/editor/contrib/inlineCompletions/ghostTextController.ts [hunk 1]
```diff
@@ -54,29 +54,41 @@ export class GhostTextController extends Disposable {
...
+				: undefined;
 	}
 
 	public shouldShowHoverAt(hoverRange: Range): boolean {
 		return this.activeController.value?.shouldShowHoverAt(hoverRange) || false;
 	}
 
 	public trigger(): void {
-		this.activeController.value?.trigger();
+		this.triggeredExplicitly = true;
```
Comment (alexdima) [inline src/vs/editor/contrib/inlineCompletions/ghostTextController.ts:79]: Shouldn't this be set to `false` somewhere when the session ends?

**Review by alexdima — APPROVED** [review]
:+1:

File: src/vs/editor/contrib/inlineCompletions/ghostTextController.ts (hunk 1 above)
Comment (hediet) [reply src/vs/editor/contrib/inlineCompletions/ghostTextController.ts:83]: This is just for the lazy initialization of the active controller (which will subscribe to the suggest widget and what not). That controller should not be disposed when the session ends. The session itself has checks that it is only triggered automatically when the settings allow it.

File: src/vs/editor/common/config/editorOptions.ts [hunk 2]
```diff
@@ -3394,12 +3399,16 @@ class EditorSuggest extends BaseEditorOption<EditorOption.suggest, InternalSugge
...
 					type: 'boolean',
-					default: defaults.suggestionPreviewExpanded,
-					description: nls.localize('suggest.suggestionPreviewExpanded', "Controls whether the suggestiion preview is expanded by default.")
+					default: defaults.showInlineCompletions,
+					description: nls.localize('suggest.showInlineCompletions', "Controls whether to show inline completions in the editor.")
+				},
+				'editor.suggest.ghostTextExpanded': {
+					type: 'boolean',
+					default: defaults.ghostTextExpanded,
+					description: nls.localize('suggest.ghostTextExpanded', "Controls whether the ghost text that is used by the suggestion preview or the inline completions is expanted by default.")
```
Comment (jogo-) [inline src/vs/editor/common/config/editorOptions.ts:3410]: Typo: expan**t**ed -> expan**d**ed

Comment (alexdima) [reply src/vs/editor/common/config/editorOptions.ts:3410]: FYI @hediet

**Pull request merged**